src/ImpurityDensities/PtypePNJunctionImpurityDensity.jl

```diff
@@ -0,0 +1,59 @@
+"""
+    struct ThermalDiffusionLithiumDensity{T <: SSDFloat} <: AbstractImpurityDensity{T}
+
+A PN junctin impurity model based on lithium thermal diffusion and constant bulk impurity density.
+The surface lithium density is saturated.
+
+ref: [Dai _et al._ (2023)](https://doi.org/10.1016/j.apradiso.2022.110638)
+ 
+## Fields
+* `Li_annealing_temperature::T`: lithium annealing temperature when the lithium is diffused into the crystal. The default value is 623 K.
+* `Li_annealing_time::T`: lithium annealing time. The default value is 18 minutes.
+* `calculate_depth2surface::Function`: the function for describing the depth to surface.
+* `bulk_imp::T`: the bulk impurity density constant.
+* `bulk/surface_imp_model::T`: there two variables will be constructed with above parameters.
+"""
+
+struct PtypePNJunctionImpurityDensity{T <: SSDFloat} <: AbstractImpurityDensity{T}
+	Li_annealing_temperature::T
+	Li_annealing_time::T
+    calculate_depth2surface::Function
```

Comment:
As also suggested before: think about making this a `Geometry` and defining functions in SSD to calculate the distance to the surface of the geometries.
```suggestion
    contact_geometry::AbstractGeometry{T}
```